### PR TITLE
Expose PackageFile to scripting

### PR DESCRIPTION
**Pull request by JimMarlowe**
This PR exposes the PackageFile object so the scripting languages can load and unload them, and examine the contents. 
I had to create a new accessor, PackageFile::GetEntryName(unsigned) because the existing PackageFile::GetEntryNames() did not want to be bound.

### CODE REVIEW

File: Source/Atomic/IO/PackageFile.h [hunk 1]
```diff
@@ -86,7 +86,20 @@ class ATOMIC_API PackageFile : public Object
     const Vector<String> GetEntryNames() const { return entries_.Keys(); }
 
     // ATOMIC BEGIN
-    
+
+    /// Return a file name in the package, for scripting use ( GetEntryNames is not bindable? )
+    const String& GetEntryName(unsigned index) const 
+    {
+        unsigned nn = 0;  // .Begin() + index syntax not supported on HashMap
+        for (HashMap<String, PackageEntry>::ConstIterator j = entries_.Begin(); j != entries_.End(); ++j)
```
Comment (JoshEngebretson) [inline Source/Atomic/IO/PackageFile.h]: ```Vector<String>``` won't be picked up currently by bindings (and in general marshaling by an index accessor method can have some benefits over copying arrays/Vectors in terms of speed and dealing with outdated data).    

How about "Return a file name in the package at the specified index" as comments do show up in documentation?

I had a bit of a think on this too, as I am not sure we can count on the HashMap iterator being ordered, though if the package file is being modified while you are iterating, you probably have bigger problems :)

File: Script/Packages/Atomic/Scene.json [hunk 2]
```diff
@@ -8,7 +8,8 @@
 	"excludes" : {
 		"Scene" : {
 			"GetComponent" : ["unsigned"],
-			"MarkReplicationDirty" : ["Node"]
+			"MarkReplicationDirty" : ["Node"],
+			"GetRequiredPackageFiles" : []
 		}
```
Comment (JoshEngebretson) [inline Script/Packages/Atomic/Scene.json]: What was the issue with GetRequiredPackageFiles?  I believe this signature should be picked up by the C# bindings (JS/TS don't have Vector instance support yet)

File: Script/Packages/Atomic/Resource.json [hunk 3]
```diff
@@ -1,11 +1,16 @@
...
 	"classes" : ["Resource", "ResourceCache", "XMLFile", "PListFile", "JSONFile", "Image", "ResourceNameIterator"],
 	"overloads": {
 		"Image": {
 			"GetPixel": ["int", "int"],
 			"SetSize": ["int", "int", "int", "unsigned"]
+		},
+		"ResourceCache": {
+			"AddPackageFile": ["String", "unsigned"],
+			"RemovePackageFile": ["String", "bool", "bool"]
 		}
```
Comment (JoshEngebretson) [inline Script/Packages/Atomic/Resource.json]: Nice 👍

File: Source/Atomic/IO/PackageFile.h (hunk 1 above)
Comment (JimMarlowe) [inline Source/Atomic/IO/PackageFile.h]: Comment changed.  The HashMap has an order-ish and the const iterating should provide valid values for each index for the current state of the PackageFile. The object could change, and this method will not guarantee an item will remain at a specific index. GetEntryNames() only gives you a pile of names without index context, so maybe index is a bad term.

File: Script/Packages/Atomic/Scene.json (hunk 2 above)
Comment (JimMarlowe) [inline Script/Packages/Atomic/Scene.json]: GetRequiredPackageFiles() was getting an error trying to convert a PackageFile pointer to an AObject something, which won't happen. So I excluded it, the loss from not having it scripted will be minimal. It is only used in the client-server scene+packagefile replication internals, which are also not exposed to scripting.

**Comment by JoshEngebretson** [comment]
Ok great, thanks for the updates, LANDING! ✈️

**Pull request merged**